torch/_meta_registrations.py

```diff
@@ -311,3 +311,22 @@ def meta_addbmm(self, batch1, batch2, *, beta=1, alpha=1):
 
 torch.library.impl(meta_lib, "addbmm")(meta_addbmm)
 torch.library.impl(meta_lib, "addbmm.out")(meta_addbmm)
+
+@torch.library.impl(meta_lib, "_cdist_forward")
+def meta_cdist_forward(x1, x2, p, compute_mode):
+    check(x1.dim() >= 2, lambda: f"cdist only supports at least 2D tensors, X1 got: {x1.dim()}D")
+    check(x2.dim() >= 2, lambda: f"cdist only supports at least 2D tensors, X2 got: {x2.dim()}D")
+    check(
+        x1.size(-1) == x2.size(-1),
+        lambda: f"X1 and X2 must have the same number of columns. X1: {x1.size(-1)} X2: {x2.size(-1)}"
+    )
+    check(utils.is_float_dtype(x1.dtype), lambda: "cdist only supports floating-point dtypes, X1 got: {x1.dtype}")
+    check(utils.is_float_dtype(x2.dtype), lambda: "cdist only supports floating-point dtypes, X2 got: {x2.dtype}")
+    check(p >= 0, lambda: "cdist only supports non-negative p values")
+    r1 = x1.size(-2)
+    r2 = x2.size(-2)
+    batch_tensor1 = x1.shape[:-2]
+    batch_tensor2 = x2.shape[:-2]
+    output_shape = list(torch.broadcast_shapes(batch_tensor1, batch_tensor2))
+    output_shape.extend([r1, r2])
+    return x1.new_empty(output_shape)
```

Comment:
The logic in the original function is more complicated. Are the two equivalent? It'd make sense to me that this logic in the meta functiion is all it's going on in the original function, but I have not checked.

Reply:
This one I worked out by just reading the docs. If the implementation disagrees it's probably a bug

Reply:
The implementation looks a bit messy. It may very well just do this, but it's not clear at first sight.